### ISSUE TITLE
Are nightlys tolerated on flathub

### ISSUE THREAD

**Issue by lovetox** [flathub/flathub#2975]
Seems some projects use Flathub for nightlys

https://github.com/flathub/org.freedesktop.Sdk.Extension.rust-nightly
https://github.com/flathub/org.gnome.Totem.Devel

Im refering to this blog post

https://blogs.gnome.org/alexl/2019/02/19/changes-in-flathub-land/

where it says beta channel is not for nightlys, i guess stable channel is then even more not for nightlys.

But this blog post is 2 years old, so has the position changed?

Thanks

**Comment by hfiguiere** [comment]
For the rust sdk it's really because it's nightly. It's automated, and used by only one project AFAIK.

Totem.Devel is not a nightly It's the devel version, and it's only on the beta branch.  Last update is 11 days old.

So no the position hasn't really change.

**Comment by hfiguiere** [comment]
If you want to provide nightlies, there is a github action

**Comment by lovetox** [comment]
So what are the rules now?

Nightlys are not allowed in the beta channel?
Nightlys are allowed in the stable channel but it should be marked accordingly for example with "-nightly" at the end or ".Devel"?

This is an action we need to set ourself on the repo?
Or an Admin sets somewhere else?
Because in the rust sdk nightly repo no action is visible currently, but that could be because im not a member of that repo.

**Comment by lovetox** [comment]
I found this https://discourse.gnome.org/t/best-practices-to-distribute-flatpak-apps-in-gnome-nightly-repository/5589/3

**Comment by hfiguiere** [comment]
> So what are the rules now?

Flathub is not for Nightly builds. Flathub is for user ready software distribution, not for continuous integration.
 
> Nightlys are not allowed in the beta channel?

Never have. The beta channel is meant to provide beta release eventually.

> Nightlys are allowed in the stable channel but it should be marked accordingly for example with "-nightly" at the end or ".Devel"?

No. The example of Totem.Devel is **beta only** and could be done on the beta channel, but the developer wanted it to be separate. Fair enough.

> This is an action we need to set ourself on the repo? Or an Admin sets somewhere else? Because in the rust sdk nightly repo no action is visible currently, but that could be because im not a member of that repo.

See details https://github.com/bilelmoussaoui/flatpak-github-actions
tl;dr it's for developers that want to provide flatpak nightlies on their own. Separate from Flathub.

**Comment by hfiguiere** [comment]
> I found this https://discourse.gnome.org/t/best-practices-to-distribute-flatpak-apps-in-gnome-nightly-repository/5589/3

GNOME Nightly is not flathub. GNOME has its own infrasctructure and repositories. So they have different policies and goals.

**Comment by lovetox** [comment]
Ok thanks, i will close this issue now and we will see if the nightly gets accepted.

you may want to update your guidelines, because the first sentence here is contradicting what was discussed here now

**Comment by hfiguiere** [comment]
things is I (myself) can't change any of these wiki pages. (tl;dr github clearly suck in that aspect)

But then reading what you linked, it clearly talks about *stable* software and not development snapshot. So it's pretty clear to me. (it doesn't mention the beta channel)

**Comment by lovetox** [comment]
I have to say communication here is very confusing.

i linked 2 projects who are clearly development snapshots. This is clearly not allowed by the current guidlines.
Yet you seem to have no problems with it, even defending it with weird arguments

- "only one project uses it"
- "its a devel version" (Yeah also called development snapshot)

still

> no position has not changed

**Comment by barthalion** [comment]
@lovetox Flathub is not built to be a continuous integration platform. We have limited infrastructure resources and cannot afford to build each commit. We make exceptions for projects with low development activity, but it requires prior discussion.
